process/transactionProcessor.go

```diff
@@ -484,6 +500,26 @@ func checkIfMoveBalanceNotarized(tx *transaction.ApiTransactionResult) bool {
 	return true
 }
 
+func (tp *TransactionProcessor) checkIfIntraShardRelayedTransaction(
```

Comment:
This checks relayer shard & user shard, but not shard of `innerTx.receiver`. If inner tx is cross-shard, but relayed one (wrapper) is intra-shard, a "false positive" success might happen (even if tx is still pending). If so, then the check maybe should be "is tx relayed & completely intra-shard, all actors in a shard".

I am not sure about this, though, so - it's just a question, not a comment.

Reply:
yup, agree, will come up with a fix for this :)